NickvisionTubeConverter.Shared/Controllers/MainWindowController.cs

```diff
@@ -172,6 +177,44 @@ public async Task StartupAsync()
                 NotificationSent?.Invoke(this, new NotificationSentEventArgs(_("Unable to unlock keyring. Restart the app to try again."), NotificationSeverity.Error));
             }
         }
+        //Check Network
+        if (!await CheckNetworkConnectivityAsync())
+        {
+            NotificationSent?.Invoke(this, new NotificationSentEventArgs(_("No active internet connection"), NotificationSeverity.Error, "no-network"));
+        }
+        NetworkChange.NetworkAvailabilityChanged += async (sender, e) =>
+        {
+            if (await CheckNetworkConnectivityAsync())
+            {
+                NotificationSent?.Invoke(this, new NotificationSentEventArgs("", NotificationSeverity.Success, "network-restored"));
+            }
+            else
+            {
+                NotificationSent?.Invoke(this, new NotificationSentEventArgs(_("No active internet connection"), NotificationSeverity.Error, "no-network"));
+            }
+        };
+    }
+
+    /// <summary>
+    /// Checks for an active network connection
+    /// </summary>
+    /// <returns>True if network connection active, else false</returns>
+    public async Task<bool> CheckNetworkConnectivityAsync()
+    {
+        foreach (var addr in new[] { "8.8.8.8", "http://www.baidu.com", "http://www.aparat.com" })
```

Comment:
Could you make the array a private readonly variable, set in the constructor, and used here? That way we avoid constantly making a new array allocation everytime we check network...